extensions/amp-social-share/0.1/validator-amp-social-share.protoascii

```diff
@@ -72,11 +54,31 @@ tags: {  # <amp-social-share>
   attrs: {
     name: "type"
     mandatory: true
-    value_regex: "(twitter|facebook|pinterest|linkedin|gplus|email)"
+  }
+  attrs: {
+    name: "data-share-endpoint"
+    value_url: {
+      allowed_protocol: "ftp"
+      allowed_protocol: "http"
+      allowed_protocol: "https"
+      allowed_protocol: "mailto"
+      # Whitelisting additional commonly observed third party
+      # protocols which should be safe
+      allowed_protocol: "sms"
+      allowed_protocol: "viber"
+      allowed_protocol: "whatsapp"
```

Comment:
This is a fairly important topic. I have two questions here:

1. /to @powdercloud / Do we currently have any validation rules `<a href="">`? Technically our case is an extension if a simple `<a>` tag.
2. /to @mkhatib / Can you test what would happen for a endpoint URL that looks like this:
```
j
a
v
a
s
r
i
p
t:
```

Ideally we'd have a blacklist here, but let's first see what are the answers to these questions.